src/tests/ftest/container/open.py

```diff
@@ -125,24 +123,22 @@ def test_container_open(self):
         # Create the pool and connect. Then create the container from the pool
         # Add the pool and the container created into a list
         container_uuids = []
-        #for i in range(2):
-        i = 0
-        while i < 2:
+        for _ in range(2):
             self.pool.append(TestPool(
                 self.context, dmg_command=self.get_dmg_command()))
             self.pool[-1].get_params(self)
             self.pool[-1].create()
             self.pool[-1].connect()
-            self.container.append(TestContainer(self.pool[-1]))
+            self.container.append(TestContainer(pool=self.pool[-1]))
             self.container[-1].get_params(self)
             self.container[-1].create()
             container_uuids.append(uuid.UUID(self.container[-1].uuid))
-            i += 1
 
         # Decide which pool handle and container UUID to use. The PASS/FAIL
         # number corresponds to the index for self.pool and container_uuids
         pool_handle_index = poh_state_num
         container_uuid_index = uuid_state_num
+        open_failed = False
```

Comment:
There is a `TestContainer.opened` (https://github.com/daos-stack/daos/blob/master/src/tests/ftest/util/test_utils_container.py#L394) property that should negate the need for this test to maintain its own flag.  This value is read during `TestWithServers.destroy_containers`: https://github.com/daos-stack/daos/blob/master/src/tests/ftest/util/apricot/apricot/test.py#L622

Reply:
I use the opened value to check if open failed and it works. Thanks for the suggestion.